### PR TITLE
Features/8069

### PR DESCRIPTION
**Pull request by sslaws**
## Status
Ready

## Fixes or Implements [AB#8069](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/8069) and [AB#8070](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/8070)
- [X] Enhancement
- [ ] Bug
- [ ] Documentation

## Description:
Completes the work on [AB#8067](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/8067) - Protective Word Cache
Implemented tasks [AB#8069](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/8069) and [AB#8070](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/8070)

## Testing
- [X] Unit Tests Updated
- [ ] Functional Tests Updated
- [ ] Not Required

### Steps to Reproduce:
N/A

### UI Changes
No

### Browsers Tested
- [ ] Chrome
- [ ] Safari
- [ ] Edge
- [ ] Firefox

Provide an explanation for any test(s) not completed.

## Notes/Additional Comments
The Object serialization and deserialization works but suggestions are welcome.

### CODE REVIEW

File: Apps/Database/src/Delegates/DBGenericCacheDelegate.cs [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+            }
+
+            this.logger.LogDebug($"Finished updating cache object in DB with result {result.Status}");
+            return result;
+        }
+
+        /// <inheritdoc />
+        public DBResult<GenericCache> DeleteCacheObject(GenericCache cacheObject, bool commit = true)
+        {
+            this.logger.LogTrace($"Deleting GenericCache object from DB...");
```
Comment (bradhead) [inline Apps/Database/src/Delegates/DBGenericCacheDelegate.cs]: minor nit: the $ isn't needed  here. Myabe it was at one point? @"" would ensure only one copy of this string is held.

**Comment by healthgatewaybc** [comment]
Immunization Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=immunization_pr_410)

**Comment by healthgatewaybc** [comment]
Patient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=patientservice_pr_410)

**Comment by healthgatewaybc** [comment]
Patient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=patientservice_pr_410)

**Comment by healthgatewaybc** [comment]
Immunization Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=immunization_pr_410)

**Comment by healthgatewaybc** [comment]
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_410)

File: Apps/Database/src/Delegates/DBGenericCacheDelegate.cs (hunk 1 above)
Comment (sslaws) [reply Apps/Database/src/Delegates/DBGenericCacheDelegate.cs]: NIT Resolved.

**Comment by healthgatewaybc** [comment]
Immunization Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=immunization_pr_410)

**Comment by healthgatewaybc** [comment]
Patient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=patientservice_pr_410)

**Comment by healthgatewaybc** [comment]
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_410)

**Pull request merged**